test/util/GTestHelpers.h

```diff
@@ -175,4 +175,22 @@ class CopyShield {
   }
 };
 
+// Helper that creates a gtest matcher from an existing one. The existing
+// matcher takes type `T`. The new matcher will accept `T`, `std::nullopt_t` and
+// `std::optional<T>`. It will match if actual and expected both have no value
+// and, otherwise, if they both have a value it will apply the given matcher on
+// the contained value.
```

Comment:
That comment is imprecise, this function does neither take nor return matchers.

It takes an explicit type `T` , and a function from `T -> Matcher<T>` (where the `T` is the expected value for the matcher, this could be implemented by a requires clause etc), 
and lifts it to a function from `optional<T> -> Matcher<optional<T>>` , by handling the case of `nullopt` as expected (nullopt matches nullopt) for both the expected and actual value.